src/core/qgstaskmanager.cpp

```diff
@@ -55,6 +57,7 @@ qint64 QgsTask::elapsedTime() const
 
 void QgsTask::start()
 {
+  mNotStartedMutex.unlock();
```

Comment:
I think you should flip this order -- there's a chance wait for finished might return in between these statements otherwise

Reply:
Yes, you right. Done